conans/test/util/detect_test.py

```diff
@@ -31,23 +31,14 @@ def detect_default_in_mac_os_using_gcc_as_default_test(self):
         if platform.system() != "Darwin":
             return
 
-        def _execute(command):
-            proc = Popen(command, shell=True, bufsize=1, stdout=PIPE, stderr=STDOUT)
-
-            output_buffer = []
-            while True:
-                line = proc.stdout.readline()
-                if not line:
-                    break
-                # output.write(line)
-                output_buffer.append(str(line))
-
-            proc.communicate()
-            return proc.returncode, "".join(output_buffer)
+        try:
+            output = subprocess.check_output(["gcc", "--version"], stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError:
+            # gcc is not installed or there is any error (no test scenario)
+            return
 
-        proc_return, output = _execute("gcc --version")
-        if proc_return != 0 or "clang" not in output:
-            # Not test scenario (gcc should display clang in output
+        if "clang" not in output:
+            # Not test scenario gcc should display clang in output
             return
```

Comment:
Mhh, may we raise here to to realize that maybe the default OSX policy for "gcc" has changed?

Reply:
I know that it is not the scope of the test, but useful to know if something changes

Reply:
Yes It's a good idea

Reply:
like: `raise Exception("Apple gcc doesn't point to clang with gcc frontend anymore! please check")`